src/screens/PoolTogether/PoolTogetherPurchase.js

```diff
@@ -151,77 +156,46 @@ class PoolTogetherPurchase extends React.Component<Props, State> {
     this.updatePurchaseFeeAndTransaction();
     logScreenView('View PoolTogether Purchase', 'PoolTogetherPurchase');
   }
 
-  updateFeeAndCheckAllowance = async () => {
+  componentDidUpdate(prevProps: Props) {
+    if (!isEqual(this.props.poolAllowance, prevProps.poolAllowance)) {
+      this.updatePurchaseFeeAndTransaction();
+    }
+  }
+
+  updateAllowanceFeeAndTransaction = () => {
     const { poolToken } = this.state;
-    const { useGasToken, poolAllowance } = this.props;
+    const { poolAllowance, estimateTransaction } = this.props;
     const hasAllowance = poolAllowance[poolToken];
     if (!hasAllowance) {
-      const {
-        txFeeInWei,
-        gasToken,
-        transactionPayload,
-      } = await getApproveFeeAndTransaction(poolToken, useGasToken) || {};
-      if (txFeeInWei) {
-        this.setState({
-          gasToken,
-          allowPayload: transactionPayload,
-          txFeeInWei,
-        });
-      }
+      const allowPayload = getApproveTransaction(poolToken);
+      const { to, data, amount } = allowPayload;
+      estimateTransaction(to, amount, data);
+      this.setState({ allowPayload });
     }
   }
 
   updatePurchaseFeeAndTransaction = () => {
     const { poolToken, numberOfTickets } = this.state;
-    const {
-      useGasToken,
-      poolAllowance,
-      baseFiatCurrency,
-      balances,
-      rates,
-    } = this.props;
+    const { poolAllowance, estimateTransaction } = this.props;
     if (poolAllowance[poolToken]) {
-      this.setState({
-        purchasePayload: null,
-      }, async () => {
-        const {
-          txFeeInWei,
-          gasToken,
-          transactionPayload,
-        } = await getPurchaseTicketFeeAndTransaction(numberOfTickets, poolToken, useGasToken) || {};
-        if (txFeeInWei) {
-          const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
-          const feeSymbol = get(gasToken, 'symbol', ETH);
-          const feeDecimals = get(gasToken, 'decimals', 'ether');
-          const feeNumeric = utils.formatUnits(txFeeInWei.toString(), feeDecimals);
-          const feeInFiat = formatFiat(parseFloat(feeNumeric) * getRate(rates, feeSymbol, fiatCurrency), fiatCurrency);
-          const feeDisplayValue = formatTransactionFee(txFeeInWei, gasToken);
-          const isDisabled = !isEnoughBalanceForTransactionFee(balances, transactionPayload);
-          const purchasePayload = {
-            transactionPayload,
-            feeInFiat,
-            feeSymbol,
-            feeDisplayValue,
-            isDisabled,
-            txFeeInWei,
-            gasToken,
-          };
-          this.setState({ purchasePayload });
-        }
+      this.setState({ purchasePayload: null }, () => {
+        const purchasePayload = getPurchaseTicketTransaction(numberOfTickets, poolToken);
+        const { to, data, amount } = purchasePayload;
+        estimateTransaction(to, amount, data);
+        this.setState({ purchasePayload });
```

Comment:
Is it worth moving:

```suggestion
        const purchasePayload = getPurchaseTicketTransaction(numberOfTickets, poolToken);
        const { to, data, amount } = purchasePayload;
        estimateTransaction(to, amount, data);
```
To a function, to avoid calling `this.setState` inside another `this.setState`?

Reply:
Previously `this.setState` callback had a promise inside, but now it looks that we can get rid of reset (set as `null`) and just go ahead with `this.setState` after new value is calculated 👍